### PR TITLE
Add build for Nix/NixOS

### PR DESCRIPTION
**Pull request by HaosGames**
This adds build instructions for [NixOS](https://nixos.org) and the flake files to go along with it. 

I found that I can omit the `libpgm-dev` and `libnorm-dev` dependency. I tried this because I didn't found those in nixpkgs. Somehow it builds and works regardless.

### CODE REVIEW

**Comment by HaosGames** [comment]
Actually: don't merge this yet
found that nix from the master branch doesn't actually fetch the sub directories which leads to a cmake error

**Comment by HaosGames** [comment]
Now it should work. 
I had to change from using `github:` to `git+https://` because `github:`uses `fetchzip`which doesn't fetch the submodules. `git+https://` however uses fetchgit which recently got changed to fetching the submodules by default.

**Comment by WeebDataHoarder** [comment]
I believe this has x86_64 hardcoded in so won't build on AArch64 installations

**Comment by HaosGames** [comment]
For now, yes. I am still relatively new to writing in the nix language. I believe I can use numtides flake-utils to change this.

**Comment by HaosGames** [comment]
Ready to merge

File: README.md [hunk 1]
```diff
@@ -81,6 +81,26 @@ make release-static -j$(nproc)
 
 Make the package: [p2pool-git](https://aur.archlinux.org/packages/p2pool-git/)
 
+### [Nix/NixOS](https://nixos.org)
+
+This is a flake only project. So you have to use [nixUnstable with nix flakes](https://nixos.wiki/wiki/Flakes) to build or install p2pool. 
+The commands below use the new flake specific reference-format, so be sure to also set `ca-refernces` in `--experimental-features`.
```
Comment (SChernykh) [inline README.md:87]: `ca-refernces` a typo?

Comment (HaosGames) [reply README.md:87]: Correct

**Review by SChernykh — APPROVED** [review]

**Pull request merged**